### ISSUE TITLE
C-c C-l cannot import modules

### ISSUE THREAD

**Issue by XStargate** [idris-hackers/idris-mode#456]
Apologies for the extreme newbie question. When I "import Effect.Random" in a Idris code file and "C-c C-l" it, it always has the error that "Can't find import Effect/Random".

I know when using Idris repl in the shell, adding '-p effects' will solve the problem. But I am not sure how I can do the similar thing to make it right in emacs. I also tried to "M-x idris-set-idris-load-packages" but it seemed not to work.

I used the spacemacs Idris layer, which I assume should be same with this one. And I used stack to install the Idris, but I am not sure if it is the reason to cause this problem.

I really appreciate it if you can help me with it. Thank you.

**Comment by david-christiansen** [comment]
I just replied to your mailing list post, but this might be a better place to discuss it.

What specifically happens when you do `M-x idris-set-idris-load-packages`? That works for me, and I know of Spacemacs users who it works for, so perhaps there's some UI thing we can clarify.

**Comment by XStargate** [comment]
Thank you for your kind reply!

When I tried `M-x idris-set-idris-load-packages`, it shows all Idris packages, including base, contrib, effects, prelude, pruviloj, as shown below:

![screenshot from 2017-07-24 21-57-44](https://user-images.githubusercontent.com/11036260/28552250-44a3d758-70bb-11e7-85fc-15d7a5189d14.png)

So following the instruction, I used `<RET>` to select the `effects` and the `effects` will disappear from the list so the package is supposed to be loaded, as shown below:

![screenshot from 2017-07-24 22-01-41](https://user-images.githubusercontent.com/11036260/28552325-b579a980-70bb-11e7-8476-6c5fa60d3159.png)

In this step, there is no error. But when I `C-c C-l` the idris file or `:module Effect.Random` in the `idris-repl`, it still has the error "Can't find import Effect/Random". So I assume I actually did not succeed in loading the "effects" packages.

BTW, I checked the `idris-interpreter-flags` and it is `nil`.
Please let me know if you need more information about it. Thank you!

**Comment by david-christiansen** [comment]
It sounds like Helm's interface is what's causing your issues here. For some reason it's hiding the instructions to provide an empty string when done picking packages, which is `C-RET` in the default Helm keybindings (I have no idea if Spacemacs has changed that).

So try picking `effects`, pressing `RET`, then pressing `C-RET` when the menu comes back. It should ask if you want to save for future sessions, which it does using a file variable.

**Comment by XStargate** [comment]
I tried your solution and It works this time ! And I am able to `C-c C-l` the Idris file. But it seems not to work in `idris-repl` though I repeated the same step in `idris-repl`.

Thank you so much for your kind help! And it took me one day to work on it.

**Comment by david-christiansen** [comment]
Glad it worked! The C-RET thing in Helm is a bit of a pain.

The REPL should inherit the settings and imports of the file that it is loaded in - it's really designed to be used in the context of a project. Have you tried `C-c C-l` in a file with the right settings, and then using the REPL from there?

**Comment by XStargate** [comment]
Yeah, I used the REPL from `C-c C-l` in a file with the right settings. But I still cannot `:module Effect.Random`, and the weird thing is I even cannot `:module Data.Vect`. As shown below:

![screenshot from 2017-07-24 22-43-59](https://user-images.githubusercontent.com/11036260/28553436-9d34f306-70c1-11e7-8f12-933e0dd5c10c.png)

And you can see "effect_test" has been checked correctly and can be loaded. But `Effect.Random` and `Data.Vect` are failed.

**Comment by abailly** [comment]
@XStargate Still relevant?

**Comment by sp1ff** [comment]
I realize this issue is from long ago, but I'd like to resurrect it. In my `Main.idr` I say `import System.Console.GetOpt`. This works fine when building via `idris2` in the shell because I added "contrib" to the depends item in my .ipkg file. 

But when, in Emacs, I try to load the file into the REPL (i.e. `idris-load-file`) I am told "Can't find import System/Console/GetOpt". I have verified that, in this buffer, `idris-load-packages` is `'("prelude" "contrib" "base")`.

Any ideas?
